src/app/song/song-search/song-search.component.ts

```diff
@@ -0,0 +1,55 @@
+import { Component, OnInit } from '@angular/core';
+import { NgForm } from '@angular/forms';
+import { SongService } from '../songs.service';
+import { PageEvent } from '@angular/material';
+import { Song, Genre } from '../song.model'
+import { Subscription } from 'rxjs';
+
+
+@Component({
+  selector: 'app-song-search',
+  templateUrl: './song-search.component.html',
+  styleUrls: ['./song-search.component.css']
+})
+export class SongSearchComponent implements OnInit {
+  selectedSong: Song;
+  songs: Song[];
+  totalSongs:number = 0;
+  songsPerPage:number = 2;
+  currentPage:number = 1;
+  // pageSizeOptions = [3,4,10];
+  private songSub: Subscription;
```

Comment:
If you copy, understand what it is for.
Implement ondestroy and unsubscribe there